Gordon360/Services/HousingService.cs

```diff
@@ -0,0 +1,774 @@
+            if (idResult != null && idResult.Any())
+            {
+                throw new ResourceCreationException() { ExceptionMessage = "An existing application ID was found for this user. Please use 'EditApplication' to update an existing application." };
+            }
+
+            //----------------
+            // Save the application editor and time
+
+            // All SqlParameters must be remade before being reused in an SQL Query to prevent errors
+            SqlParameter timeParam = new SqlParameter("@NOW", now);
+            editorParam = new SqlParameter("@EDITOR_ID", editorID);
+
+            // If an existing application was not found for this editor, then insert a new application entry in the database
+            int? newAppResult = context.Database.ExecuteSqlCommand("INSERT_AA_APPLICATION @NOW, @EDITOR_ID", timeParam, editorParam); //run stored procedure
+            if (newAppResult == null)
+            {
+                throw new ResourceCreationException() { ExceptionMessage = "The application could not be saved." };
+            }
+
+            //----------------
+            // Retrieve the application ID number of this new application
+
+            // The following is ODD, I know. It seems you cannot execute the same query with the same sql parameters twice.
+            // Thus, these two sql params must be recreated after being used in the last query:
+
+            idResult = null;
+
+            // All SqlParameters must be remade before each SQL Query to prevent errors
+            timeParam = new SqlParameter("@NOW", now);
+            editorParam = new SqlParameter("@EDITOR_ID", editorID);
+
+            idResult = RawSqlQuery<ApartmentAppIDViewModel>.query("GET_AA_APPID_BY_NAME_AND_DATE @NOW, @EDITOR_ID", timeParam, editorParam); //run stored procedure
+            if (idResult == null)
+            {
+                throw new ResourceNotFoundException() { ExceptionMessage = "The new application ID could not be found." };
+            }
+
+            int applicationID = idResult.FirstOrDefault().AprtAppID;
+
+            //----------------
+            // Save applicant information
+
+            SqlParameter appIDParam = null;
+            SqlParameter idParam = null;
+            SqlParameter programParam = null;
+
+            foreach (ApartmentApplicantViewModel applicant in apartmentApplicants)
+            {
+                // All SqlParameters must be remade before being reused in an SQL Query to prevent errors
+                appIDParam = new SqlParameter("@APPLICATION_ID", applicationID);
+                idParam = new SqlParameter("@ID_NUM", applicant.StudentID);
+                if (applicant.OffCampusProgram != null)
+                {
+                    programParam = new SqlParameter("@APRT_PROGRAM", applicant.OffCampusProgram);
+                }
+                else
+                {
+                    programParam = new SqlParameter("@APRT_PROGRAM", "");
+                }
```

Comment:
Some nullish coalescing would make this much cleaner:
```suggestion
                    programParam = new SqlParameter("@APRT_PROGRAM", applicant.OffCampusProgram ?? "");
                }
```